src/core/atomic.d

```diff
@@ -621,36 +619,32 @@ else
 
 private
 {
-    version (AsmX86)
+    bool atomicValueIsProperlyAligned(T)(ref T val) pure nothrow @nogc @trusted
+    {
+        return atomicPtrIsProperlyAligned(&val);
+    }
+
+    bool atomicPtrIsProperlyAligned(T)(T* ptr) pure nothrow @nogc @safe
     {
         // NOTE: Strictly speaking, the x86 supports atomic operations on
         //       unaligned values.  However, this is far slower than the
         //       common case, so such behavior should be prohibited.
-        bool atomicValueIsProperlyAligned(T)(ref T val) pure nothrow @nogc @trusted
+        static if (T.sizeof > size_t.sizeof)
         {
-            return atomicPtrIsProperlyAligned(&val);
-        }
-
-        bool atomicPtrIsProperlyAligned(T)(T* ptr) pure nothrow @nogc @safe
-        {
-            // NOTE: 32 bit x86 systems support 8 byte CAS, which only requires
-            //       4 byte alignment, so use size_t as the align type here.
-            static if (T.sizeof > size_t.sizeof)
+            version (X86)
+            {
+                // cmpxchg8b only requires 4-bytes alignment
                 return cast(size_t)ptr % size_t.sizeof == 0;
+            }
             else
+            {
+                // e.g., x86_64 cmpxchg16b requires 16-bytes alignment
                 return cast(size_t)ptr % T.sizeof == 0;
```

Comment:
@ibuclaw: Is that a 'good' default logic for non-x86 targets? I think it's at least better than the current `return true` for non-DMD...